web-app/js/portal/data/GeoNetworkRecordFetcher.js

```diff
@@ -38,11 +38,20 @@ Portal.data.GeoNetworkRecordFetcher = Ext.extend(Ext.util.Observable, {
             var metadataRecord = store.getAt(0);
             var dataCollection = Portal.data.DataCollection.fromMetadataRecord(metadataRecord);
 
-            _this.dataCollectionStore.add(dataCollection);
-            Ext.MsgBus.publish(PORTAL_EVENTS.VIEW_DATA_COLLECTION, dataCollection);
+            if (!metadataRecord) {
+                _this._errorLoadingDataCollection(uuid);
+            }
+            else {
+                _this.dataCollectionStore.add(dataCollection);
+                Ext.MsgBus.publish(PORTAL_EVENTS.VIEW_DATA_COLLECTION, dataCollection);
+            }
```

Comment:
Can I suggest we swap this condition around? I think it'd be nicer because then the condition won't be negated and the happy-path behaviour comes first.